### ISSUE TITLE
Heat fluxes

### ISSUE THREAD

**Issue by AndyHoggANU** [adele157/easterlies-collaborative-project#22]
How do they change?

**Comment by adele157** [comment]
Also freshwater/salt input changes? i.e. All the buoyancy forcing.

**Comment by AndyHoggANU** [comment]
Also, the different contributions to heat/sw fluxes -- to separate circumpolar role of evaporation vs localised role of enhanced sea ice fluxes??

**Comment by matthew-england-unsw** [comment]
Yes, looking at the animations yesterday of SSH changes - and seeing the Ekman sign reversal appearing to be so rapidly circumpolar - it looks like either the DSW changes are propagating super quick via planetary waves, or there are separate wind-driven buoyancy effects in the non-DSW regions.  Due to a conspiracy of wind changes giving “same sign” buoyancy tendencies via E-P, net heat fluxes, combined with sea-ice advection.   Interested to see how these analyses turn out!   The non-DSW region wind anomaly experiment is another way to tackle this.

**Comment by AndyHoggANU** [comment]
Piecing together heat/FW flux changes now ... Will report back once I have something.

**Comment by AndyHoggANU** [comment]
So, I have a few updates on the heat flux code. It's not making a lot of sense to me, but I thought I would put it out there and maybe we can make some progress on Thursday. There is a script in a pull request which @adele157 can merge at her leisure.

The first thing to note is that net, shelf-averaged heat and FW fluxes appear to equilibrate after a year or so and follow a pretty regular cycle.

![image](https://user-images.githubusercontent.com/26753100/130353172-e72de982-434a-498f-8223-0eeb2aa1c977.png)

**Comment by AndyHoggANU** [comment]
The annual cycle and the anomalies are plotted here, averaged over the last 5 years. But what doesn't make sense to me is that the ice FW contribution (which I thought should be `(melt - sfc_salt_flux_ice)`) is larger than the seasonal cycle of net FW flux. Furthermore, the ice flux makes more sense to me than the net FW flux, so maybe that is wrong?

![image](https://user-images.githubusercontent.com/26753100/130353284-bc042ddd-25d9-4f40-87f5-97845d2f2350.png)

**Comment by AndyHoggANU** [comment]
Given the strong and repeating seasonal cycle, I thought it best to break the data up into summer vs winter to show maps. Here are the net heat fluxes which show strong cooling on the shelf in tehe UP case ... but this is mainly in summer, not winter. I guess that is influenced by ice, but it wasn't what I expected.

![image](https://user-images.githubusercontent.com/26753100/130353418-b0a166b6-b374-4b66-9b43-9fd8d8867d16.png)

**Comment by AndyHoggANU** [comment]
Same for net FW. Again, the biggest anomalies are are in summer ...

![image](https://user-images.githubusercontent.com/26753100/130353446-8f84c4da-9078-4e68-840e-4daa5efd57ee.png)

**Comment by AndyHoggANU** [comment]
And finally, here are the FW flux anomalies due to just ice. These show the strong negative FW flux that I would expect in winter in the UP case.

![image](https://user-images.githubusercontent.com/26753100/130353583-7cd8eb7c-e6f9-45f0-bc12-78e6dc6c7d88.png)
 
All in all, I'm not sure these analyses resolve any of our questions. While the winter ice formation in the UP case drives the salinification that leads to our increased DSW formation, that signal is not present in the either the net heat or FW fluxes ... Any suggestions to resolve this please let me know.

**Comment by matthew-england-unsw** [comment]
Cool, thanks heaps Andy.   A summer maximum imprint in ice-free conditions makes sense maybe?  My hypothesis that the applied circumpolar wind anomalies drive net heat and FW flux anomalies that “double up” in terms of density anomalies (via wind-driven evaporative cooling causing surface T-S to both cool and salinify) would work more pervasively over open ocean regions?   Not when covered by sea-ice?  So during summer and/or over coastal polynyas?   That would then reduce the level of stratification that needs to be punched through during winter to create deep mix layers.

**Comment by AndyHoggANU** [comment]
I have updated the heat/FW flues notebook and merged. The take-home message is that ice does indeed dominate the FW budget and that the UP case is characterised by a year-round net negative FW flux anomaly. This FW flux anomaly occurs in both DSW formation regions AND more generally around the shelf. 

The implication is that perhaps a major driver of all the dynamics we see here is a general increase in sea ice formation and export. Next job here is to diagnose the ice formation and export from the shelf.

**Comment by AndyHoggANU** [comment]
I worked just a little on ice fluxes today. I would have liked to quantify the actual flux of ice across the 1000m isobath, but I'm not convinced that our monthly diagnostics of `hi` and `vvel` would give us a valid result. (We could still try that as a cross-check but it may not be necessary.)

Instead, I have just looked at net area-integrated melt over the shelf (converted to Sv). Since the only sink of ice is melt and offshore export, this gives us the net export. See figure.


![Unknown-1](https://user-images.githubusercontent.com/26753100/132430130-9b20036b-5438-45d8-9f99-87c4fe7e74f3.png)

**Comment by AndyHoggANU** [comment]
The main point here is that we have a robust net negative melt (export of ice) in the UP case (and the opposite in the DOWN case). So, does this progress our chicken vs egg argument? I think so ... the increase in flux could either be (a) because there is more ice being produced due to katabatics, and thicker ice advected offshore at the same velocity, or (b) because ice is exported at a greater rate than previously. But Paul has already looked at ice thickness and velocity, where we see thinner ice and faster export( see https://github.com/adele157/easterlies-collaborative-project/issues/10). So, I think (b) is our only option.

To summarise -- I am proposing that the key determinant here is increase export of ice across the continental shelf due to increased northerlies, which leads to thinner ice and more DSW production ...

**Comment by adele157** [comment]
Ok, I think I agree. So in the UP case, there is more sea ice formation in winter, more export of sea ice northwards, but despite this less melting in summer over the shelf. That was one of our hypotheses before, that the warming SST in the UP case might be driving the thinning of sea ice. But this seems to show that's false, is that right?

**Comment by AndyHoggANU** [comment]
Yes, I think I would argue that the thinning is driven by export. But note that the timeseries I am showing are from melt alone, so we can't infer from this when the actual export would occur. For this we would need to either estimate from monthly fields -- or perhaps run an extra year with additional ice diagnostics. Is that worthwhile?

**Comment by matthew-england-unsw** [comment]
Nicely synthesized in a sentence @AndyHoggANU - "To summarise -- I am proposing that the key determinant here is increase export of ice across the continental shelf due to increased northerlies, which leads to thinner ice and more DSW production ..."   Agreed, this gels with what I think was about my 13th hypothesis for what was going on in these experiments.   :-)  
Thanks for all the extra analyses everyone and see you tomorrow.....

**Comment by adele157** [comment]
@StephenGriffies here is the anomaly in salinity restoring compared with the net buoyancy flux anomaly from all freshwater flux terms. It is presented as a climatology over years 5-10 of the perturbations and is integrated over the continental shelf. I will add a sentence in the paper saying that the change in restoring is negligible.

<img width="400" alt="Screen Shot 2022-08-19 at 6 13 41 am" src="https://user-images.githubusercontent.com/8506963/185486626-da344ba4-5858-4079-a091-b38af38c6c4d.png">

**Comment by StephenGriffies** [comment]
Beautiful!  

I assume that is not just zero? Right?

**Comment by adele157** [comment]
Yes, zoomed in values here:
<img width="369" alt="Screen Shot 2022-08-19 at 8 34 45 am" src="https://user-images.githubusercontent.com/8506963/185506976-2d07069b-27c1-4e2d-ad32-466aa30e87a6.png">

The sign of the restoring is also consistent with what would be expected from the [SSS change](https://github.com/adele157/easterlies-collaborative-project/issues/4#issuecomment-875259285).

**Comment by StephenGriffies** [comment]
Thanks for pursuing this.  In many models, SSS restoring is the dominant player in the NAtl, but it is great to see it is negligible for the Antarctic shelf.  

I wonder: is the SSS restoring turned off under sea ice?  There is an option in MOM5 do to that:  

salt_restore_under_ice=.false.  

That could help explain why there is so tiny an SSS restore here.

**Comment by matthew-england-unsw** [comment]
Is there a unit issue here (e.g. as per thread with Pedro's experiments)?
Normally perturbations in circulation like this would result in SSS changes that might see a restoring kickback of the order 10-20% of the typical budget terms?  This look like tiny fractions of a single percentage?

**Comment by adele157** [comment]
No, I think the restoring term is really small here because we're not directly perturbing the freshwater budget, i.e. the restoring change occurs indirectly through the wind forcing impact on the sea ice. The 10-20% offset in restoring occurs when we do really large hosing perturbations.

**Comment by matthew-england-unsw** [comment]
Great, thanks for confirming. 👍🙏

**Comment by StephenGriffies** [comment]
Also, could someone please check to see if salt_restore_under_ice=.false., which would further account for the small SSS restoring contribution.

**Comment by AndyHoggANU** [comment]
Just checked:
`salt_restore_under_ice = .true.`

**Comment by StephenGriffies** [comment]
ok, thanks.  I guess it is just a very weak restoring for the reasons @adele157 mentioned.

**Comment by matthew-england-unsw** [comment]
Yes must be so.  Thanks all.   I am kinda surprised NA studies can find that SSS restoring sometimes dominates the salt budget, whilst here over the shelf it’s tiny.  But happy to close this case and move on, of course.

**Comment by StephenGriffies** [comment]
Sorry, but I too am having a tough time convincing myself that SSS restoring is totally negligible.  I can see it being a few 10s of percent, but many orders smaller than fresh water contribution prompts me to ask one more question, again about units.  

When computing the contribution of salt fluxes to buoyancy evolution, one needs to compute the buoyancy time tendency as 

\partial b/\partial t = -g/(rho0 *dz) * beta * Qs

where dz = thickness of top grid cell, beta is haline contraction coefficient, and Qs is salt flux.  Beta is computed as 

beta = (1/rho0)*\partial \rho/\partial S

For the restoring salt flux diagnosed in MOM5, Qs has units kg salt/(m^2 sec), in which case we need beta measured in units of inverse of salt concentration rather than inverse salinity. There is a factor of 1000 difference. That is, beta should have values around 0.8 rather than 8e-4.  

Do you agree?

**Comment by adele157** [comment]
Yes, agreed Steve. I will recalculate!

**Comment by adele157** [comment]
Here are the correct buoyancy fluxes. The salinity restoring is included in the 'Freshwater' term and is also plotted with its own line as well. 

The 'Freshwater' term is calculated as:
beta * (SSS * pme_river - 1000 * (sfc_salt_flux_ice + sfc_salt_flux_restore)) 
and is integrated over the shelf area.

<img width="800" alt="Screen Shot 2022-08-23 at 8 45 44 pm" src="https://user-images.githubusercontent.com/8506963/186140529-9c69055f-fddf-4589-892d-26dafa2ff428.png">

Note I haven't changed the axes labels here from what Andy had before, I think this is actually kg/s.

In summary: Our sea ice mechanism dominating the buoyancy flux change still holds. Though perhaps we should add in a sentence saying that the salinity restoring offsets the sea ice buoyancy change by X%.

Thanks so much @StephenGriffies for picking up this error!

**Comment by StephenGriffies** [comment]
Thanks for the recalculation @adele157 .   Things make more sense now, and very glad the larger restoring term is still not dominating the sea ice mechanism.  

Note that for units on buoyancy flux, we might wish to bring things to more conventional units.  Namely, kg/s is a mass transport unit not a buoyancy flux.  If multiply by g/(rho0*dz) then the units will be m/s^3, which is the units for a buoyancy time tendency.  Not a big deal either way.  But if keeping the current numbers, then explain the details in the caption, since kg/s is not a conventional unit for "buoyancy flux".

**Comment by adele157** [comment]
Good suggestion, I will convert to a proper buoyancy flux.
